### ISSUE TITLE
Dealing with streams

### ISSUE THREAD

**Issue by murrayju** [nodejs/nan#774]
Does nan provide any helpers for dealing with node's Readable streams? I need to connect such a stream to an InputStream interface in a native library. I've detailed what I've tried and what problems I have in this stack overflow post: https://stackoverflow.com/q/50472159/866224

It feels like this should be a common use case. Any advice would be much appreciated.

**Comment by kkoopa** [comment]
You can probably adapt this: https://nodeaddons.com/streaming-data-into-a-node-js-c-addon/

**Comment by murrayju** [comment]
@kkoopa thanks for the suggestion. I evaluated this before, and it seems like a square peg in a round hole sort of solution. What I need is high throughput of a binary data stream. The `streaming-worker` approach would involve multiple copies of the data, and framing/unframing into "messages". Also the native library that I'm using manages its own threads, and doesn't fit well into the "worker" model.

I thought it would be better to take a more direct approach, where I could directly copy data from the node Buffer into the native library's buffer. I'm using the same mechanisms from the source of `AsyncWorker` to sync with the event loop using libuv.

While I could try to shoehorn this into `streaming-worker`, I'm concerned that the performance will be bad, and I'm skeptical that it might even have the same problem. This would also be a significant effort to attempt. I'd be much happier if someone with more intimate knowledge of the node (stream) internals could tell me what's wrong with my solution. Either the streams are getting stuck in some bad state, or the node/js code isn't getting a chance to execute, and I'd like to understand why.

**Comment by murrayju** [comment]
After digging into this much more, I discovered that this is very likely a client side issue (though I'm not sure why). While I saw this problem when making 4+ connections from the same node client process, if I instead spawned a new client process for each connection, I saw no such issues (tried up to 100 clients).

For browser based clients, there's a known limitation of 6 TCP connections, so I was likely running into this. Not sure why node clients would have a similar limitation.

Thanks @kkoopa for your help!
